File: C++/Level2/타겟 넘버.cpp
#include <string>
#include <vector>

using namespace std;

void dfs(vector<int> numbers, int target, int lev, int sum, int* count) {

	if (lev == numbers.size()) {
		if (sum == target) (*count)++;
		return;
	}

	dfs(numbers, target, lev + 1, sum + numbers[lev], count);
	dfs(numbers, target, lev + 1, sum - numbers[lev], count);
}

int solution(vector<int> numbers, int target) {

	int answer = 0;

	dfs(numbers, target, 0, 0, &answer);

	return answer;
}

File: programmers-coding-test/Python/Level2/타겟 넘버.py
def target_number(lev, value, numbers, target):
    count = 0
    if lev == len(numbers):
        if value == target:
            count += 1
        return count
    count += target_number(lev + 1, value + numbers[lev], numbers, target)
    count += target_number(lev + 1, value - numbers[lev], numbers, target)
    return count


def solution(numbers, target):
    return target_number(0, 0, numbers, target)
